### ISSUE TITLE
Reusing translations from other projects?

### ISSUE THREAD

**Issue by nrenner** [nrenner/brouter-web#306]
PR #304 introduces dynamic translations of highway, surface and smoothness tag values. Some with underscore were added manually to the `en.json` and got removed by `i18next`:

https://github.com/nrenner/brouter-web/blob/f37489cf265ab5e4b24224ac135ed1d66cc351eb/locales/en.json#L196-L207

They would need to be added to [keys.js](https://github.com/nrenner/brouter-web/blob/master/locales/keys.js), but I left them off for now, as I think we should not translate them ourselves, but ideally reuse existing translations, some OSM editor must already have those.

On Transifex we are in the OpenStreetMap organisation, and e.g. the Vespucci editor is also part of that.

Can translations be shared between projects on Transifex (haven't done any research yet)?

**Comment by nrenner** [comment]
The concepts for reusing translations on Transifex I found are [Sharing Translation Memory](https://docs.transifex.com/translation-memory/sharing-tm) and [Glossary](https://docs.transifex.com/glossary/glossary). These are presented as suggestions, so still invole translators. Translation Memory also allows to automatically translate matching texts, but I'm a bit sceptical about that.

I haven't found anything that would allow to reference or link to specific translation keys from other projects.

i18next has the concept of [Namespaces](https://www.i18next.com/principles/namespaces) to load from separate files.

So my idea would be extract the required keys directly from the translation files of another project into separate files and not go through Transifex.

Some candidates are:
- [Beautified JOSM preset](https://github.com/simonpoole/beautified-JOSM-preset): [PO format](https://github.com/simonpoole/beautified-JOSM-preset/tree/master/i18n), (preset: [master_preset.xml](https://github.com/simonpoole/beautified-JOSM-preset/blob/master/master_preset.xml)), [smoothness only as descriptions](https://github.com/simonpoole/beautified-JOSM-preset/blob/d77ccd9475e07645c15e8bc1cb09d71289273dd7/master_preset.xml#L35-L46) ¹
- [iD](https://github.com/openstreetmap/iD), [YAML source format](https://github.com/openstreetmap/iD/blob/develop/data/presets.yaml), [JSON out format](https://github.com/openstreetmap/iD/tree/develop/dist/locales), [smoothness only as descriptions](https://github.com/openstreetmap/iD/blob/9d70ad4119f744fd3620ff5c19776f9e675f0c19/data/presets.yaml#L2339-L2361) ¹, surface values not translated ([iD#2708](https://github.com/openstreetmap/iD/issues/2708))?
- JOSM is complicated: [binary format](https://josm.openstreetmap.de/browser/josm/trunk/resources/data?order=name), PO format download from [Launchpad](https://translations.launchpad.net/josm/trunk/) involves registration and receiving a download link as email.
- parsing wiki pages (e.g. [Key:surface](https://wiki.openstreetmap.org/wiki/Key:surface#Values))?

¹ `smoothness` shares some values with `trail_visibility` (has direct value translations), but not `very_bad`, `very_horrible` and `impassable`

None of the candidates is ideal, other ideas?

**Comment by Bibi56** [comment]
> None of the candidates is ideal, other ideas?

Don't parse the wiki, use the corresponding [Data Items](https://wiki.openstreetmap.org/wiki/Data_items)?

**Comment by nrenner** [comment]
Thanks for the hint, Data Items would be better than parsing, but probably depends on coverage.

The general problem with the wiki is, that there often are no explicit, single-word translations of the values, only descriptions. E.g. the [Template:DE:Map Features:surface](https://wiki.openstreetmap.org/wiki/Template:DE:Map_Features:surface) page has value translations only as bold highlight within the description. And the data items only sometimes seem to have a tag translation, e.g. [surface=sett](https://wiki.openstreetmap.org/wiki/Item:Q16171).

**Comment by Bibi56** [comment]
You're right: it depends on coverage but if we start to use it more, the coverage will get improved. If somebody is missing a translation he/she can improve the Wiki. Way easier than to translate in many tools in many formats.
Yes, some people destroyed some wiki pages for values as the values were in the main page... it would have been better to keep them.
